hack/lib/golang.sh

```diff
@@ -865,26 +862,19 @@ kube::golang::build_binaries() {
     # These are "local" but are visible to and relied on by functions this
     # function calls.  They are effectively part of the calling API to
     # build_binaries_for_platform.
-    local goflags goldflags goasmflags gogcflags gotags
+    local goflags goldflags gogcflags gotags
 
-    # This is $(pwd) because we use run-in-gopath to build.  Once that is
-    # excised, this can become ${KUBE_ROOT}.
-    local trimroot # two lines to appease shellcheck SC2155
-    trimroot=$(pwd)
-
-    goasmflags="all=-trimpath=${trimroot}"
```

Comment:
was this just incorrect before? I thought we *wanted* to trim the root but leave relative source paths in place... those should still be reproducible, right?

Reply:
From testing it looks like `-trimpath` does automatically what we were trying to do manually.

```
$ make kubectl DBG=1
+++ [1115 08:39:32] Building go targets for linux/amd64
    k8s.io/kubernetes/cmd/kubectl (static)
thockin-glaptop5 kubernetes f72e3d build_trimpath /$ strings _output/bin/kubectl | grep 'k8s\.io.*\.go'$ | head -10
/home/thockin/src/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/spf13/pflag/bool.go
/home/thockin/src/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/spf13/pflag/bool_slice.go
/home/thockin/src/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/spf13/pflag/bytes.go
/home/thockin/src/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/spf13/pflag/count.go
/home/thockin/src/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/spf13/pflag/duration.go
/home/thockin/src/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/spf13/pflag/duration_slice.go
/home/thockin/src/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/spf13/pflag/flag.go
/home/thockin/src/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/spf13/pflag/float32.go
/home/thockin/src/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/spf13/pflag/float32_slice.go
/home/thockin/src/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/spf13/pflag/float64.go

$ make kubectl
$ strings _output/bin/kubectl | grep 'k8s\.io.*\.go'$ | head -10
k8s.io/apimachinery@v0.0.0/pkg/util/sets/string.go
k8s.io/apimachinery@v0.0.0/pkg/util/sets/set.go
k8s.io/klog/v2@v2.110.1/internal/severity/severity.go
k8s.io/klog/v2@v2.110.1/internal/buffer/buffer.go
k8s.io/klog/v2@v2.110.1/internal/clock/clock.go
k8s.io/klog/v2@v2.110.1/internal/dbg/dbg.go
k8s.io/klog/v2@v2.110.1/internal/serialize/keyvalues.go
k8s.io/klog/v2@v2.110.1/internal/serialize/keyvalues_slog.go
k8s.io/klog/v2@v2.110.1/internal/sloghandler/sloghandler_slog.go
k8s.io/klog/v2@v2.110.1/exit.go
```

Note, it also changes paths into modules, but that seems OK?  If you need to debug (good old dlv!) then you need to map source files manually.  It probably chages the output of logging when the full caller is logged (rare).

Also note:

```
$ strings _output/bin/kubectl | grep thockin
```

No output on the non-DBG

Reply:
More simply:

```
$ make kubectl
+++ [1115 08:45:53] Building go targets for linux/amd64
    k8s.io/kubernetes/cmd/kubectl (static)

$ strings _output/bin/kubectl | grep 'kubectl\.go'
k8s.io/kubernetes/cmd/kubectl/kubectl.go

$ make kubectl DBG=1
+++ [1115 08:46:16] Building go targets for linux/amd64
    k8s.io/kubernetes/cmd/kubectl (static)

$ strings _output/bin/kubectl | grep 'kubectl\.go'
/home/thockin/src/kubernetes/_output/local/go/src/k8s.io/kubernetes/cmd/kubectl/kubectl.go
```